src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -308,18 +308,23 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                     if (!refreshAll)
                     {
-                        applicationSettings = new Dictionary<string, ConfigurationSetting>(_applicationSettings, StringComparer.OrdinalIgnoreCase);
+                        mappedData = new Dictionary<string, ConfigurationSetting>(_mappedData, StringComparer.OrdinalIgnoreCase);
                         watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>(_watchedSettings);
 
                         foreach (KeyValueChange change in keyValueChanges.Values.Concat(changedKeyValuesCollection))
                         {
                             if (change.ChangeType == KeyValueChangeType.Deleted)
                             {
-                                applicationSettings.Remove(change.Key);
+                                mappedData.Remove(change.Key);

```

Comment:
We are running mappers on all KeyValueChanges before updating `_watchedSettings`. That means we are not creating deep copies of the settings in _watchedSettings dictionary when refreshAll is false.